src/test/java/hw6/steps/UserTablePageSteps.java

```diff
@@ -0,0 +1,81 @@
+package hw6.steps;
+
+import hw6.MapTable;
+import hw6.WebDriverSingleton;
+import hw6.pages.UserTablePage;
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.DataTableType;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class UserTablePageSteps {
+
+    UserTablePage userTablePage = new UserTablePage(WebDriverSingleton.INSTANCE.getDriver());
+
+    @DataTableType
+    public MapTable defineTableColumns(DataTable dataTable) {
+        return new MapTable(dataTable.transpose().asLists().stream()
+                .collect(
+                        Collectors.toMap(
+                                column -> column.get(0),
+                                column -> column.subList(1, column.size())
+                        )
+                )
+        );
+    }
```

Comment:
Why do you decide create `MapTable` class instead of `User` for the data?
In this case step 
```java
@Then("User table should contain following values:")
    public void userTableShouldContainValues(MapTable table) {
        assertEquals(userTablePage.getIndexes(), table.getTableValues("Number"));
        assertEquals(userTablePage.getUsernames(), table.getTableValues("User"));
        assertEquals(userTablePage.getDescriptionTexts(), table.getTableValues("Description"));
    }
```
could be look like 
```java
@Then("User table should contain following values:")
    public void userTableShouldContainValues(User table) {
        assertEquals(userTablePage.getUserData(), table);
    }
```